Dockerfile

```diff
@@ -1,14 +1,122 @@
-FROM rust:alpine as builder
-RUN apk add clang musl-dev openssl-dev cmake make
-
-COPY . /tmp
-WORKDIR /tmp
-
-RUN cargo --version
-RUN cargo build --release
-
-FROM alpine as base
-COPY --from=builder /tmp/target/release/convco /usr/bin/convco
-
-ENTRYPOINT [ "convco" ]
+# syntax = docker/dockerfile-upstream:master-labs
+#-*-mode:dockerfile;indent-tabs-mode:nil;tab-width:2;coding:utf-8-*-
+# vi: ft=dockerfile tabstop=2 shiftwidth=2 softtabstop=2 expandtab:
+FROM alpine:3.14 AS upx-downloader
+SHELL ["/bin/ash", "-o", "pipefail", "-c"]
+RUN                                                                                                                          \
+  apk add --no-cache                                                                                                         \
+    curl=7.78.0-r0                                                                                                           \
+    jq=1.6-r1                                                                                                                \
+    xz=5.2.5-r0 ;
+ARG REPO="upx/upx"
```

Comment:
Is `upx` necessary? Although it will make the binary smaller it adds more complexity and dependencies to the build.

Reply:
It makes it much smaller; the docker image after upx compression with `scratch` base layer is less than 2MB; that's the power of UPX. If you don't think size matters, then feel free to drop the compression layer altogether.